Modules/Module_Processing/Module_ASL/xASL_wrp_Quantify.m

```diff
@@ -205,7 +207,40 @@ function xASL_wrp_Quantify(x, PWI4D_Path, pathOutputCBF, M0Path, SliceGradientPa
 	x = rmfield(x, 'hematocrit');
 end
 
+% b. We model the expected hematocrit from age & sex
+% only if the Hematocrit was not specifically provided
+if ~isfield(x, 'Hematocrit')
+    IndexSetsAge = find(strcmpi(x.S.SetsName, 'age'));
+    IndexSetsSex = find(strcmpi(x.S.SetsName, 'sex'));
+    if ~isempty(IndexSetsAge) || ~isempty(IndexSetsSex)
+        fprintf('%s\n', 'Trying to infer hematocrit from age & sex');
+        
+        age = x.S.SetsID(:, IndexSetsAge);
+        
+        % Convert sex correctly
+        sex = x.S.SetsID(:, IndexSetsSex);
+        sexOptions = x.S.SetsOptions{:, IndexSetsSex};
+        sexN = zeros(length(sex), 1)
```

Comment:
We set everything to 0, then set some values to 1 or 2. And in the end set everything that isn't 1 or 2 to NaN.

It's much easier to set sexN to NaNs here at 223. And then remove line 231.

Reply:
Fixed in b7f4265a627f48c78dae39af4356a335ff27a3f8